pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -260,11 +139,15 @@ func MergedAdvancedDeployment(atlasDeploymentAsAtlas mongodbatlas.AdvancedCluste
 		for k, v := range replicationSpec.RegionConfigs {
 			// the response does not return backing provider names in some situations.
 			// if this is the case, we want to strip these fields so they do not cause a bad comparison.
-			if v.BackingProviderName == "" {
+			if v.BackingProviderName == "" && k < len(mergedDeployment.ReplicationSpecs[i].RegionConfigs) {
```

Comment:
why did we need to add this comparison? what does it mean?

Reply:
We used to fix misconfiguration and fill gaps in runtime. As I moved some of them to the validation step I had to add some checks to avoid nil pointer and/or out of bound indexes